src/llmq/quorums_instantsend.cpp

```diff
@@ -788,6 +803,16 @@ bool CInstantSendManager::PreVerifyInstantSendLock(const llmq::CInstantSendLock&
         }
     }
 
+    if (islock.IsDeterministic()) {
+        LOCK(cs_main);
+        const auto llmqType = Params().GetConsensus().llmqTypeInstantSend;
+        const auto dkgInterval = GetLLMQParams(llmqType).dkgInterval;
+        const auto blockIndex = LookupBlockIndex(islock.cycleHash);
+        if (blockIndex == nullptr || blockIndex->nHeight % dkgInterval != 0) {
+            return false;
+        }
```

Comment:
This might be problematic.. We likely don't want to p2p ban a peer just because we don't have this block yet...

Reply:
I changed it to slightly penalize a peer in order to avoid the case when some peer spams an invalid cycleHash.